### PR TITLE
Changed cookies to use cookie-rs

### LINKED ISSUE #26: Use cookie-rs for cookie parsing
Use https://docs.rs/cookie/ for cookie parsing

### PR DESCRIPTION
**Pull request by Osrepnay**
Should resolve #26 and #15.

### CODE REVIEW

File: src/cookie.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    ///
+    /// ```
+    /// use octane::prelude::*;
+    /// use cookie::Cookie as CookieRs;
+    ///
+    /// let cookie_rs_cookie = CookieRs::new("name", "value");
+    /// let octane_cookie = octane::cookie::Cookie::from(cookie_rs_cookie);
+    /// println!("{:?}", octane_cookie);
+    /// ```
+    pub fn from<'a>(cookie: CookieRs<'a>) -> Cookie<'a> {
```
Comment (Daksh14) [inline src/cookie.rs]: Use the from trait here instead of an function https://doc.rust-lang.org/std/convert/trait.From.html

File: src/cookie.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        cookies_vec
+    }
+    // Prepare the `Set-Cookie` Header string from the values
+    // in the HashMap
+    pub(crate) fn serialise(&self) -> String {
+        format!("Set-Cookie: {}", self.cookie.to_string())
+    }
+}
+
+impl Eq for Cookie<'_> {}
```
Comment (Daksh14) [inline src/cookie.rs:92]: You can put this in the derive macro

**Review by Daksh14 — CHANGES_REQUESTED** [review]

File: src/cookie.rs (hunk 1 above)
Comment (Osrepnay) [reply src/cookie.rs]: Fixed.

File: src/cookie.rs (hunk 2 above)
Comment (Osrepnay) [reply src/cookie.rs:92]: I've already tried that, can't because Cookie in cookies-rs doesn't have the Eq trait.

**Pull request merged**